### ISSUE TITLE
Tutorial: set writeControl to adjustableRunTime

### ISSUE THREAD

**Issue by MakisH** [precice/openfoam-adapter#34]
With `writeControl runTime`, sometimes we get non-round write times. Set this to `adjustableRunTime` (and check if it also relates to #26).

**Comment by derekrisseeuw** [comment]
I can uderline that the writeControl works identical in openfoam plus. Normally OpenFOAM will just select the closest time to the nearest writeTime. With adjustableRunTime you force openFOAM to adapt the timestep and write at exactly the specified writeTime. 

However, I also noticed that even with adjustableRunTime, non-round times can be written. This happens if openfoam needs to increase the timePrecision. Then it will write times such as:

`9.80000000000008`

**Comment by MakisH** [comment]
> I can uderline that the writeControl works identical in openfoam plus. Normally OpenFOAM will just select the closest time to the nearest writeTime. With adjustableRunTime you force openFOAM to adapt the timestep and write at exactly the specified writeTime.

This is correct. Do you mean hear that you approve or disapprove this change?
 
> However, I also noticed that even with adjustableRunTime, non-round times can be written. This happens if openfoam needs to increase the timePrecision. Then it will write times such as:
> 
> `9.80000000000008`

This is very interesting! In which scenario would it have to increase the timePrecision? Do you maybe have an easy-to-share setup at the moment where this is observed?

**Comment by derekrisseeuw** [comment]
> This is correct. Do you mean hear that you approve or disapprove this change?
With this I mean that I approve of the change. I think it is better to force the adapter to use the exact writetimes. 

> This is very interesting! In which scenario would it have to increase the timePrecision?
I was running a transient case with only the fluid side. For some reason the timePrecision had to be increased. So this is related to this issue:
[](https://bugs.openfoam.org/view.php?id=815)

I have a working example with boundary conditions from groovyBC. Do you have Swak4Foam installed? Then I can share the case.

**Comment by MakisH** [comment]
I don't have Swak4Foam on this computer, but maybe it's a good time to install it!

**Comment by derekrisseeuw** [comment]
I'll prepare the fluid case to share it

**Comment by derekrisseeuw** [comment]
In:
`$FOAM_SRC/OpenFOAM/db/Time/Time.C`
You can see how openfoam handles the timeprecision. On the part where it checks whether the new time directory has a diffferent format it can increase the time-precision, which leads to this behaviour.

**Comment by derekrisseeuw** [comment]
> I'll prepare the fluid case to share it

You can find the tutorial at the gitlab!
